spec.bs

```diff
@@ -494,6 +506,218 @@ The <dfn for=Navigator method>runAdAuction(|config|)</dfn> method steps are:
+
+       : [=fenced frame reporting metadata/direct seller is seller=]
+       :: true
+
+       Otherwise (i.e., if there was a component auction), a struct with the following members:
+       : [=fenced frame reporting metadata/fenced frame reporting map=]
+       :: a [=map=] «[ "<code>buyer</code>" → «», "<code>seller</code>" → «»,
+          "<code>component-seller</code>" → «»]»
+
+       : [=fenced frame reporting metadata/direct seller is seller=]
+       :: false
+
+    : [=fenced frame reporting metadata/visibility=]
+    :: "<a for=visibility>`opaque`</a>"
+1. Set |pendingConfig|'s [=fenced frame config/on navigate callback=] to a closure of
+   [=asynchronously finish reporting=] partially evaluated with a reference to |pendingConfig|'s
+   [=fenced frame config/fenced frame reporting metadata=]'s
+   [=fenced frame reporting metadata/value=]'s
+   [=fenced frame reporting metadata/fenced frame reporting map=].
+1. Set |pendingConfig|'s [=fenced frame config/nested configs=]'s [=nested configs/value=] to
+   the result of running [=create nested configs=] with |winningBid|'s
+   [=generated bid/ad component descriptors=].
+1. Return |pendingConfig|.
+
+</div>
+
+<div algorithm="asynchronously finish reporting">
+
+To <dfn>asynchronously finish reporting</dfn> given a reference to a
+[=fencedframetype/fenced frame reporting map=] |reportingMap| and [=leading bid info=]
+|leadingBidInfo|:
+1. TODO: [=Finalize a reporting destination=] with |reportingMap|
+
+</div>
+
+<div algorithm="create nested configs">
+
+To <dfn>create nested configs</dfn> given [=generated bid/ad component descriptors=]
+|adComponentDescriptors|:
+1. Let |nestedConfigs| be an empty [=list=] «».
+1. If |adComponentDescriptors| is null:
+  1. Return |nestedConfigs|.
+1. [=set/For each=] |adComponentDescriptor| in |adComponentDescriptors|:
+  1. Let |fencedFrameConfig| be a [=fenced frame config=] with the following members:
+    : [=fenced frame config/mapped url=]
+    :: a struct with the following members:
+      : [=mapped url/value=]
+      :: |adComponentDescriptor|'s [=ad descriptor/url=]
+
+      : [=mapped url/visibility=]
+      :: "<a for=visibility>`opaque`</a>"
+
+    : [=fenced frame config/container size=]
+    :: null
+
+    : [=fenced frame config/content size=]
+    :: If |adComponentDescriptor|'s [=ad descriptor/size=] is null, then null. Otherwise, a struct
+       with the following members:
+       : [=content size/value=]
+       :: |adComponentDescriptor|'s [=ad descriptor/size=] TODO: Resolve screen-relative sizes and
+          macros and cast this properly.
+
+       : [=content size/visibility=]
+       :: "<a for=visibility>`opaque`</a>"
+
+    : [=fenced frame config/interest group descriptor=]
+    :: null
+
+    : [=fenced frame config/on navigate callback=]
+    :: null
+
+    : [=fenced frame config/effective sandbox flags=]
+    :: a struct with the following members:
+      : [=effective sandbox flags/value=]
+      :: TODO

```

Comment:
ditto

Reply:
subsequent PR (sandbox flags haven't been fleshed out as much as permissions in the FF spec), so e.g. it isn't really specified what this field should include (sandboxed features? unsandboxed features?)

Reply:
oh, you meant about struct/items